notifications/management/commands/send_notifications.py

```diff
@@ -274,288 +308,198 @@ def find_bill_search_updates(self, search_params):
+                person__id=person_id
+            )
+            if new_sponsorships.count() > 0:
+                person_update = base_update_dct.copy()
+                person = councilmatic_models.Person.objects.get(id=person_id)
+                person_update['New sponsorships'].update({
+                    'name': person.name,
+                    'slug': person.slug,
+                })
+                for sponsorship in new_sponsorships:
+                    person_update['New sponsorships']['bills'].append({
+                        'identifier': sponsorship.bill.identifier,
+                        'slug': sponsorship.bill.slug,
+                        'description': sponsorship.bill.title,
+                    })
+                person_updates.append(person_update)
+        return person_updates
 
-
-            for bill_slug, bill_group in itertools.groupby(group, key=inner_grouper):
-
-                bill_group = list(bill_group)
-
-                bill = {
-                    'identifier': bill_group[0][2],
-                    'slug': bill_group[0][3],
-                    'description': bill_group[0][4],
-                    'actions': []
+    def find_committee_action_updates(self, committee_ids, minutes=15):
+        committee_updates = []
+        for committee_id in committee_ids:
+            new_actions = councilmatic_models.BillAction.objects.annotate(
+                action_date=Cast(
+                    Coalesce(
+                        Case(
+                            When(date__exact='', then=None),
+                            When(date__isnull=False, then='date'),
+                            default=None,
+                            ouput_field=django_models.CharField()
+                        ),
+                        Value(str(self.get_threshold(minutes+1)))
+                    ),
+                    django_models.DateTimeField()
+                )
+            ).filter(
+                organization__id=committee_id,
+                action_date__gte=self.get_threshold(minutes)
+            )
+            if new_actions.count() > 0:
+                committee = councilmatic_models.Organization.objects.get(
+                    id=committee_id
+                )
+                committee_update = {
+                    'name': committee.name,
+                    'slug': committee.slug,
+                    'bills': {}
                 }
-
-
-                for row in sorted(bill_group, key=lambda x: x[7], reverse=True):
-                    action = {
-                        'description': row[5],
-                        'date': row[6]
-                    }
-
-                    bill['actions'].append(action)
-
-                committee_group['bills'].append(bill)
-
-            committee_updates.append(committee_group)
-
+                for action in new_actions.distinct(
+                    'date', 'organization__id', 'bill__id', 'id'
```

Comment:
yes, i think in general the distincts in the queries were because of the joins, e.g., joining event to event participant would return a row for every participant -> duplicate events. i feel ok about omitting them here!